app/zap.js

```diff
@@ -0,0 +1,217 @@
+    this._registerIpcListeners()
+
+    // Show the window as soon as the application has finished loading.
+    this.mainWindow.webContents.on('did-finish-load', async () => {
+      this.mainWindow.show()
+      this.mainWindow.focus()
+      mainLog.timeEnd('Time until app is visible')
+      mainLog.time('Time until we know the run mode')
+
+      Promise.resolve(this.mode)
+        .then(mode => {
+          const timeUntilWeKnowTheRunMode = mainLog.timeEnd('Time until we know the run mode')
+          return setTimeout(() => {
+            if (mode === 'external') {
+              // If lnd is already running, create and subscribe to the Lightning grpc object.
+              this.startGrpc()
+              this.sendMessage('successfullyCreatedWallet')
+            } else {
+              // Otherwise, start the onboarding process.
+              this.sendMessage('startOnboarding')
+              mainLog.timeEnd('Time until onboarding has started')
+            }
+          }, timeUntilWeKnowTheRunMode < this.splashScreenTime ? this.splashScreenTime : 0)
+        })
+        .catch(mainLog.error)
+    })
+
+    this.mainWindow.on('closed', () => {
+      this.mainWindow = null
+
+      // shut down zap when a user closes the window
+      app.quit()
+    })
+  }
+
+  /**
+   * Send a message to the main window.
+   * @param  {string} msg message to send.
+   * @param  {[type]} data additional data to acompany the message.
+   */
+  sendMessage(msg, data) {
+    mainLog.info('Sending message to renderer process: %o', { msg, data })
+    this.mainWindow.webContents.send(msg, data)
+  }
+
+  /**
+   * Create and subscribe to the Lightning grpc object.
+   */
+  startGrpc() {
+    mainLog.info('Starting gRPC...')
+    try {
+      const { lndSubscribe, lndMethods } = lnd.initLnd()
+
+      // Subscribe to bi-directional streams
+      lndSubscribe(this.mainWindow)
+
+      // Listen for all gRPC restful methods
+      ipcMain.on('lnd', (event, { msg, data }) => {
+        lndMethods(event, msg, data)
+      })
+
+      this.sendMessage('grpcConnected')
+    } catch (error) {
+      dialog.showMessageBox({
+        type: 'error',
+        message: `Unable to connect to lnd. Please check your lnd node and try again: ${error}`
+      })
+      app.quit()
+    }
+  }
+
+  /**
+   * Create and subscribe to the WalletUnlocker grpc object.
+   */
+  startWalletUnlocker() {
+    mainLog.info('Starting wallet unlocker...')
+    try {
+      const walletUnlockerMethods = lnd.initWalletUnlocker()
+
+      // Listen for all gRPC restful methods
+      ipcMain.on('walletUnlocker', (event, { msg, data }) => {
+        walletUnlockerMethods(event, msg, data)
+      })
+
+      this.sendMessage('walletUnlockerStarted')
+    } catch (error) {
+      dialog.showMessageBox({
+        type: 'error',
+        message: `Unable to start lnd wallet unlocker. Please check your lnd node and try again: ${error}`
+      })
+      app.quit()
```

Comment:
I think this pattern of dialog + quit results in the dialog disappearing shortly after revealed.

Reply:
In my environment the dialog blocks the application until you hit ok, after which the application quits.